darts/tests/test_timeseries.py

```diff
@@ -165,6 +165,31 @@ def test_integer_range_indexing(self):
         # getting index for idx should return i s.t., series[i].time == idx
         assert series.get_index_at_point(16) == 3
 
+    def test_grom_group_dataframe(self):
+        # checks that the time_index is of RangeIndex type when the time_col is a list/Rangeindex
+        group = ["a", "a", "a", "b", "b", "b"]
+        values = np.random.randn(len(group))
+
+        # for time as a list
+        time = [i for i in range(len(group))]
+        df = pd.DataFrame({
+            "group": group,
+            "time": time,
+            "x": values,
+        })
+        ts = TimeSeries.from_group_dataframe(df, group_cols="group", time_col="time")
+        assert isinstance(ts[0].time_index, pd.RangeIndex)
+
+        # for time as RangeIndex
+        time = pd.RangeIndex(len(group))
+        df = pd.DataFrame({
+            "group": group,
+            "time": time,
+            "x": values,
+        })
+        ts = TimeSeries.from_group_dataframe(df, group_cols="group", time_col="time")
+        assert isinstance(ts[0].time_index, pd.RangeIndex)
+
```

Comment:
Can you add a test for `time` made out of `pd.Timestamps`, just to make sure they are converted to the expected type (`DatetimeIndex`)?

You could also try to leverage the `pytest.parametrize` functionality to make the test a bit shorter.

Also, would be nice to add a case where the index (either list of int or list of Timestamps) are not ordered, to verify that the sorting of the index behave as expected.

Reply:
Thanks for the comments and the corrections, much appreciated. There are some more tests on the group_from function inside "test_timeseries_static_covariates" already, but I agree, it's better to test and unsorted index (and datetimeindex with timestamps).